src/Model/User/Handler/RegisterUserHandler.php

```diff
@@ -28,18 +31,41 @@
     private $userCollection;
 
     /**
+     * @var ChecksUniqueUsersEmailAddress
+     */
+    private $checksUniqueUsersEmailAddress;
+
+    /**
      * @param UserCollection $userCollection
+     * @param ChecksUniqueUsersEmailAddress $checksUniqueUsersEmailAddress
      */
-    public function __construct(UserCollection $userCollection)
-    {
+    public function __construct(
+        UserCollection $userCollection,
+        ChecksUniqueUsersEmailAddress $checksUniqueUsersEmailAddress
+    ) {
         $this->userCollection = $userCollection;
+        $this->checksUniqueUsersEmailAddress = $checksUniqueUsersEmailAddress;
     }
 
     /**
      * @param RegisterUser $command
      */
     public function __invoke(RegisterUser $command)
     {
+        if ($userId = ($this->checksUniqueUsersEmailAddress)($command->emailAddress())) {
+            if (!$user = $this->userCollection->get($userId)) {
+                throw UserNotFound::withUserId($userId);
+            }
+
+            $user->registerAgain($command->name());
+
```

Comment:
No. 

For any aggregate pulled from a collection events are recorded. Only when you create a new aggregate it should be added to the repository.